### PR TITLE
6281: Add task list

### PR DESCRIPTION
**Pull request by kat1906**
Activity: AB#6281
Task: AB#6372

Screenshot 2020-05-05 at 16 42 56

### CODE REVIEW

File: app/components/general/components/bc-task-list/macro.test.js [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+  it('should render a link if item in items array has href', componentTester(setup, (harness) => {
+    harness.request(mockContext, ($) => {
+      expect($('li[data-test-id="task-0"] a[data-test-id="item-0-description"]').length).toEqual(1);
+      expect($('li[data-test-id="task-0"] a[data-test-id="item-0-description"]').text().trim()).toEqual(mockContext.params.taskList[0].items[0].description);
+      expect($('li[data-test-id="task-0"] a[data-test-id="item-1-description"]').length).toEqual(0);
+      expect($('li[data-test-id="task-1"] a[data-test-id="item-0-description"]').length).toEqual(0);
+      expect($('li[data-test-id="task-1"] a[data-test-id="item-1-description"]').length).toEqual(1);
+      expect($('li[data-test-id="task-1"] a[data-test-id="item-1-description"]').text().trim()).toEqual(mockContext.params.taskList[1].items[1].description);
+    });
+    // item-{{itemLoop.index0}}-description
```
Comment (alibombaye) [inline app/components/general/components/bc-task-list/macro.test.js:64]: comment here still needed?

File: app/components/general/components/bc-task-list/macro.njk [hunk 2]
```diff
@@ -0,0 +1,30 @@
+{% macro bcTaskList(params) %}
+  <ol data-test-id="task-list" class="bc-c-task-list">
+    {% for task in params.taskList %}
+      {% set taskLoop = loop %}
```
Comment (alibombaye) [inline app/components/general/components/bc-task-list/macro.njk:4]: what's the purpose of this? why can't we just use `task`

Comment (kat1906) [reply app/components/general/components/bc-task-list/macro.njk:4]: to remember the outer loop to be able to use loop.index in the inner loop

Comment (alibombaye) [reply app/components/general/components/bc-task-list/macro.njk:4]: 👍

**Pull request merged**